source/common/router/upstream_request.cc

```diff
@@ -814,7 +814,7 @@ void UpstreamRequestFilterManagerCallbacks::resetStream(
   // tell the router the stream reset, and let the router make the decision to
   // send a local reply, or retry the stream.
   if (reset_reason == Http::StreamResetReason::LocalReset &&
-      transport_failure_reason != "codec_error") {
+      !absl::StrContains(transport_failure_reason, "codec_error")) {
```

Comment:
I wonder if we want to protect this by the runtime guard as well in case there is some unexpected behavior change associated with the switch from != to !contains?

Reply:
done